packages/monitor-v2/src/monitor-polymarket/common.ts

```diff
@@ -680,6 +683,56 @@ export function decodeMultipleValuesQuery(decodedAncillaryData: string): Multipl
   return json;
 }
 
+export interface UMAAIRetry {
+  id: string;
+}
+export interface UMAAIRetriesLatestResponse {
+  elements: UMAAIRetry[];
+  next_cursor: string | null;
+  has_more: boolean;
+  total_count: number;
+  total_pages: number;
+}
+interface AIRetryLookupResult {
+  deeplink?: string;
+}
+
+export async function fetchLatestAIDeepLink(
+  proposal: OptimisticPriceRequest,
+  params: MonitoringParams,
+  logger: typeof Logger
+): Promise<AIRetryLookupResult> {
+  try {
+    const response = await params.httpClient.get<UMAAIRetriesLatestResponse>(params.aiApiUrl, {
+      params: {
+        limit: 50,
+        search: proposal.proposalHash,
+        last_page: false,
+        project_id: params.aiProjectId,
+      },
+      headers: {
+        "Content-Type": "application/json",
+        Authorization: `Bearer ${params.aiApiKey}`,
+      },
+    });
+
+    const result = response.data?.elements?.[0] ?? null;
```

Comment:
this would return wrong element if the search returned multiple proposals that were submitted in the same tx. For now we would have to filter them by question ID, but maybe the API itself can be changed to expose tx/questionId search combo